docs/en/advanced_config/bootloader_secure_boot.md

```diff
@@ -0,0 +1,323 @@
+with open('private_key.bin', 'wb') as f:
+    f.write(private_bytes)
+
+with open('public_key.bin', 'wb') as f:
+    f.write(public_bytes)
+```
+
+#### Installing Public Keys on Device
+
+Public keys must be installed in the device's keystore. The exact method depends on your board's keystore implementation:
+
+- Some boards use One-Time Programmable (OTP) memory
+- Some boards use secure flash storage
+- Some implementations use a stub keystore for development/testing
+
+Refer to your board's keystore implementation in `src/drivers/stub_keystore/` or board-specific keystore drivers.
+
+### Firmware Signing Process
+
+The firmware signing process involves:
+
+1. **Build firmware**: Build your PX4 firmware as usual
+2. **Create TOC**: Generate the Table of Contents structure
+3. **Sign firmware**: Sign the firmware image with your private key
+4. **Append signature**: Add the signature to the firmware image according to TOC structure
+5. **Flash to device**: Upload the signed firmware to the device
+
+::: info
+The exact tooling for firmware signing depends on your board and security requirements. Custom scripts are typically needed for production use.
+:::
+
+### TOC Creation Example
+
+The TOC structure typically looks like this:
+
+```
+[TOC Start Magic]
+[TOC Version]
+[TOC Entry 0: Firmware]
+  - Name: "APP"
+  - Start: <firmware start address>
+  - End: <firmware end address>
+  - Signature Index: 1
+  - Key Index: 0
+  - Flags: TOC_FLAG1_BOOT | TOC_FLAG1_CHECK_SIGNATURE
+[TOC Entry 1: Signature]
+  - Name: "SIG"
+  - Start: <signature start address>
+  - End: <signature end address>
+  - Flags: 0
+[TOC End Magic]
+[Firmware Binary Data]
+[Signature Data]
+```
+
+## Security Considerations
+
+### Best Practices
+
+1. **Key Management**:
+   - Use hardware security modules (HSM) for key storage when possible
+   - Never commit private keys to version control
+   - Use separate keys for development and production
+   - Implement key rotation policies
+
+2. **Build Security**:
+   - Sign firmware in a secure, controlled environment
+   - Verify signatures before distribution
+   - Use checksums for firmware integrity verification
+   - Maintain an audit log of signed firmware versions
+
+3. **Device Security**:
+   - Use OTP or secure flash for storing public keys
+   - Consider using R&D certificates for development devices (allows unsigned boot)
+   - Implement secure firmware update mechanisms
+   - Monitor for signature verification failures
+
+### Development vs Production
+
+For development, you can use the R&D Certificate feature:
+
+```c
+// In TOC, mark entry with RDCT flag
+toc_entry.flags1 |= TOC_FLAG1_RDCT;
+
+// R&D certificate capabilities
+#define RDCT_CAPS0_ALLOW_UNSIGNED_BOOT 0x1
+```
+
+R&D certificates allow specific devices (identified by UUID) to boot unsigned firmware, useful during development without compromising production security.
+
+::: warning
+Never deploy R&D certificates to production devices!
+:::
```

Comment:
Again, is any of this true? 
Is there an R&D certificate, and if so, how does it work?